core/src/main/java/com/datastax/oss/driver/api/core/type/codec/TypeCodecs.java

```diff
@@ -78,39 +98,171 @@
   public static final PrimitiveIntCodec INT = new IntCodec();
   public static final PrimitiveLongCodec BIGINT = new BigIntCodec();
   public static final PrimitiveShortCodec SMALLINT = new SmallIntCodec();
+
+  /**
+   * A codec that handles Apache Cassandra(R)'s timestamp type and maps it to Java's {@link
+   * Instant}, using the system's {@linkplain ZoneId#systemDefault() default time zone} as its
+   * source of time zone information. If you need a different time zone, consider other constants in
+   * this class, or call {@link #timestampAt(ZoneId)} instead.
+   *
+   * @see #TIMESTAMP_UTC
+   * @see #timestampAt(ZoneId)
+   */
   public static final TypeCodec<Instant> TIMESTAMP = new TimestampCodec();
```

Comment:
I'm following the established pattern of proposing 2 constants and 1 method for groups of related timestamp codecs:

1. Constant for system default zone: `FOO_SYSTEM`
2. Constant for UTC: `FOO_UTC`
3. Factory method for other zones: `fooAt`

Following this logic the constant `TIMESTAMP` should have been named `TIMESTAMP_SYSTEM` but it's too late for that.